felix/collector/dpstatshelper.go

```diff
@@ -57,8 +57,12 @@ func New(
 	goldmaneAddr := configParams.FlowLogsGoldmaneServer
 	if goldmaneAddr != "" {
 		log.Infof("Creating Flow Logs GoldmaneReporter with address %v", goldmaneAddr)
-		gd := goldmane.NewReporter(goldmaneAddr)
-		dispatchers[FlowLogsGoldmaneReporterName] = gd
+		gd, err := goldmane.NewReporter(goldmaneAddr)
+		if err != nil {
+			log.WithError(err).Error("Failed to create Flow Logs GoldmaneReporter, skipping adding reporter.")
```

Comment:
I think this is a fatal error, no?

Reply:
Currently it's up to you whether or not you want this to be fatal. Do we crash loop node if the reporter can't be configured?

I want to say "yes" but that's not how this is currently working (this error is swallowed and not even logged https://github.com/projectcalico/calico/pull/9942/files#diff-909a2a8475061194f60bb3e6864203a1fa0390d37f6071fe27c2506f382dd45eL52-L55).

Reply:
idk, maybe one for @mazdakn

Reply:
yeah

Reply:
Although I almost prefer the current behavior... presumably it panics if it hits this? 

A single error log and then nothing is pretty hard to spot. Maybe let's just make this a Fatal()

Reply:
I don't think it panics currently though, it just returns. Could you point out where that would happen?

Reply:
Oh are you referring to a send on the closed channel or something like that?

Reply:
something like that

Reply:
an accidental panic, not an intentional one

Reply:
I don't think it actually panics though, the channel isn't closed when it returns.

So even as is, this change stops the caller of Push from being blocked because the channel is full.

Reply:
My point is that we should panic if the gRPC config is invalid so that the user finds out - whether that's via an explicit panic (ideally) or an implicit one (trying to access null) I don't really care

Reply:
OK so I'll just change this so that 1) logs Fatal here and 2) the channel is properly closed when it should be in the client so a panic is forced.

That said, my points weren't that I disagree with what you're saying and that we shouldn't do that, I just wasn't intending to change / fix any bugs with these changes (ideally I didn't want bug fixing scope creep in this PR).

Reply:
Yeah, I get the desired to not add scope creep but if we don't fix these things now when else are we going to? Takes more time to raise a backlog issue than it does to switch Error to Fatal

Reply:
Ya I hear ya.